### PR TITLE
Fix for story [YONK-415]: Error occurs in users-courses-detail api

### PR DESCRIPTION
**Pull request by ayesha-baig**
@ziafazal

### CODE REVIEW

File: edx_solutions_api_integration/users/views.py [hunk 1]
```diff
@@ -970,7 +970,10 @@ def get(self, request, user_id, course_id):
             course_descriptor,
             field_data_cache,
             course_key)
-        response_data['position'] = course_module.position
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: Nit: Can we simplify it to?
```
response_data['position'] = getattr(course_module, 'position', None)
```

File: edx_solutions_api_integration/users/tests.py [hunk 2]
```diff
@@ -2189,3 +2189,25 @@ def test_mark_notification_as_read(self):
 
         # then verify unread count, which should be 0
         self.assertEqual(get_notifications_count_for_user(user_id, filters={'read': False}), 0)
+
+    @mock.patch("edx_solutions_api_integration.users.views.module_render.get_module_for_descriptor")
+    def test_user_courses_detail_get_undefined_course_module(self, mock_get_module_for_descriptor):
+        course_id = unicode(self.course.id)
+
+        # Enroll test user in test course
+        test_uri = '{}/{}/courses'.format(self.users_base_uri, self.user.id)
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: We can simplify this 
```
CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
```

File: edx_solutions_api_integration/users/tests.py [hunk 3]
```diff
@@ -2189,3 +2189,25 @@ def test_mark_notification_as_read(self):
...
+        data = {'course_id': course_id}
+
+        response = self.do_post(test_uri, data)
+        self.assertEqual(response.status_code, 201)
+
+        # Get user course details when course_module is None
+        mock_get_module_for_descriptor.return_value = None
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: I think we should not mock `get_module_for_descriptor`. we should test with actual scenario which is getting course details of a user without completing any modules.

**Comment by ayesha-baig** [comment]
@ziafazal: suggested changes have been integrated.

**Comment by ziafazal** [comment]
@ayesha-baig LGTM 👍 . Can we close this PR and create a new one against `api-input-validation` branch which we would later merge into master. That would allow us to keep API input validations separate from Eucalyptus modifications.

**Pull request merged**